### PR TITLE
Handle machine deletion when NLB/LB has been deleted

### LINKED ISSUE #347: Cluster stuck in inconsistent state when nlb is deleted out of lifecycle.
**What happened**:
User manually deleted the nlb hosting the control plane of the cluster from console. Deletion of this cluster is stuck in an inconsistent state as nlb reconciler is always hitting a 404 not found when searching for this nlb. 

**What you expected to happen**:
Cluster should not be stuck in 'deleting' phase, it could reconcile the nlb as deleted and go forward with deletion.

**How to reproduce it (as minimally and precisely as possible)**:
Create a cluster and delete the nlb-apiserver from console, then delete the cluster.

**Anything else we need to know?**:
Logs:
kubectl get events -A | grep oke-amaaaaaakdrwrhiamr4gonrrnvrts6bs
oke                               14m         Normal    FailureDomainsReady          ocicluster/oke-amaaaaaakdrwrhiamr4gonrrnvrts6bs                               FailureDomain is in ready state
oke                               48m         Warning   ReconcileError               ocicluster/oke-amaaaaaakdrwrhiamr4gonrrnvrts6bs                               failed to reconcile Api Server Network Loadbalancer: Error returned by NetworkLoadBalancer Service. Http Status Code: 404. Error Code: NotAuthorizedOrNotFound. Opc request id: . Message: Authorization failed or requested resource not found....
oke                               31m         Warning   ReconcileError               ocicluster/oke-amaaaaaakdrwrhiamr4gonrrnvrts6bs                               failed to reconcile Api Server Network Loadbalancer: Error returned by NetworkLoadBalancer Service. Http Status Code: 404. Error Code: NotAuthorizedOrNotFound. Opc request id: . Message: Authorization failed or requested resource not found....
oke                               14m         Warning   ReconcileError               ocicluster/oke-amaaaaaakdrwrhiamr4gonrrnvrts6bs                               failed to reconcile Api Server Network Loadbalancer: Error returned by NetworkLoadBalancer Service. Http Status Code: 404. Error Code: NotAuthorizedOrNotFound. Opc request id: . Message: Authorization failed or requested resource not found.

**Environment:**

- CAPOCI version: v0.11.5
- Cluster-API version (use `clusterctl version`):
- Kubernetes version (use `kubectl version`):
- Docker version (use `docker info`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by shyamradhakrishnan**
**What this PR does / why we need it**:
Handle machine deletion when NLB/LB has been deleted

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #347

### CODE REVIEW

File: cloud/scope/machine.go [hunk 1]
```diff
@@ -609,6 +609,10 @@ func (m *MachineScope) ReconcileDeleteInstanceOnLB(ctx context.Context) error {
 			LoadBalancerId: loadbalancerId,
 		})
 		if err != nil {
+			if ociutil.IsNotFound(err) {
+				m.Logger.Info("LB has been deleted")
```
Comment (joekr) [inline cloud/scope/machine.go:613]: NIT: log out the LB id?

**Review by joekr — APPROVED** [review]

**Comment by shyamradhakrishnan** [comment]
```
[SynchronizedAfterSuite] PASSED [0.000 seconds]
[SynchronizedAfterSuite]
/home/ubuntu/go/src/github.com/oracle/cluster-api-provider-oci/test/e2e/e2e_suite_test.go:261
------------------------------
[SynchronizedAfterSuite] PASSED [686.429 seconds]
[SynchronizedAfterSuite]
/home/ubuntu/go/src/github.com/oracle/cluster-api-provider-oci/test/e2e/e2e_suite_test.go:261

  Timeline >>
  STEP: Tearing down the management cluster @ 02/27/24 06:41:11.521
  << Timeline
------------------------------

Ran 7 of 28 Specs in 3274.494 seconds
SUCCESS! -- 7 Passed | 0 Failed | 0 Pending | 21 Skipped
```

**Comment by shyamradhakrishnan** [comment]
unit test 
```
$ make test
GOBIN=/home/ubuntu/go/src/github.com/oracle/cluster-api-provider-oci/hack/tools/bin ./scripts/go_install.sh sigs.k8s.io/controller-runtime/tools/setup-envtest setup-envtest v0.0.0-20230131074648-f5014c077fc3
kube-builder assets: /home/ubuntu/.local/share/kubebuilder-envtest/k8s/1.24.2-linux-amd64
KUBEBUILDER_ASSETS="/home/ubuntu/.local/share/kubebuilder-envtest/k8s/1.24.2-linux-amd64" go test -coverprofile=coverage.out ./...
?   	github.com/oracle/cluster-api-provider-oci	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/config	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/metrics	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/api/v1beta1	53.928s	coverage: 23.7% of statements
ok  	github.com/oracle/cluster-api-provider-oci/api/v1beta2	0.046s	coverage: 15.6% of statements
ok  	github.com/oracle/cluster-api-provider-oci/cloud/ociutil	0.013s	coverage: 19.6% of statements
?   	github.com/oracle/cluster-api-provider-oci/cloud/scope/mocks	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/base	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/base/mock_base	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/compute	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/compute/mock_compute	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/computemanagement	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/computemanagement/mock_computemanagement	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/containerengine	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/containerengine/mock_containerengine	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/identity	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/identity/mock_identity	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/loadbalancer	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/loadbalancer/mock_lb	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer/mock_nlb	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/vcn	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/vcn/mock_vcn	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/feature	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/version	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/cloud/scope	232.072s	coverage: 75.1% of statements
ok  	github.com/oracle/cluster-api-provider-oci/cloud/util	0.084s	coverage: 61.1% of statements
ok  	github.com/oracle/cluster-api-provider-oci/controllers	26.858s	coverage: 58.7% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/api/v1beta1	13.308s	coverage: 15.3% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/api/v1beta2	0.016s	coverage: 4.8% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/controllers	0.387s	coverage: 56.3% of statements
go tool cover -func=coverage.out -o coverage.txt
go tool cover -html=coverage.out -o coverage.html
```

File: cloud/scope/machine.go (hunk 1 above)
Comment (shyamradhakrishnan) [reply cloud/scope/machine.go]: fixed

**Review by joekr — APPROVED** [review]

**Pull request merged**